eslint-plugin-query: exhaustive-deps does not suport queryFn() {} syntax

### Describe the bug

Usage of the following syntax does not trigger an error from `"@tanstack/query/exhaustive-deps"`:

```typescript
const id = "foo";

useQuery({
  queryKey: ["deps"],
  queryFn() {
      return id;
  }
});
```

What I have found is this syntax trigger an error:


```typescript
const id = "foo";

useQuery({
  queryKey: ["deps"],
  queryFn: () => {
      return id;
  }
});
```

### Your minimal, reproducible example

https://codesandbox.io/p/devbox/fancy-lake-hymxyk

### Steps to reproduce

1. Go to https://codesandbox.io/p/devbox/fancy-lake-hymxyk
2. Fork
3. Open a new Terminal
4. Run `yarn eslint src/index.jsx`

### Expected behavior

Expected output:

```bash
yarn run v1.22.19
$ /workspace/node_modules/.bin/eslint src/index.jsx

/workspace/src/index.jsx
  25:5  error  The following dependencies are missing in your queryKey: endpoint  @tanstack/query/exhaustive-deps

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### Platform

Any

### Tanstack Query adapter

react-query

### TanStack Query version

v5.28.6

### TypeScript version

v5.3.0

### Additional context

_No response_

--- a/packages/eslint-plugin-query/src/__tests__/exhaustive-deps.test.ts
+++ b/packages/eslint-plugin-query/src/__tests__/exhaustive-deps.test.ts
@@ -726,5 +726,24 @@ ruleTester.run('exhaustive-deps', rule, {
         },
       ],
     },
+    {
+      name: 'should fail if queryFn is invalid while using FunctionExpression syntax',
+      code: normalizeIndent`
+        const id = 1;
+    
+        useQuery({
+            queryKey: [],
+            queryFn() {
+              Promise.resolve(id)
+            }
+        })
+      `,
+      errors: [
+        {
+          messageId: 'missingDeps',
+          data: { deps: 'id' },
+        },
+      ],
+    },
   ],
 })

--- a/packages/eslint-plugin-query/src/rules/exhaustive-deps/exhaustive-deps.rule.ts
+++ b/packages/eslint-plugin-query/src/rules/exhaustive-deps/exhaustive-deps.rule.ts
@@ -55,7 +55,10 @@ export const rule = createRule({
           scopeManager === null ||
           queryKey === undefined ||
           queryFn === undefined ||
-          queryFn.value.type !== AST_NODE_TYPES.ArrowFunctionExpression
+          !ASTUtils.isNodeOfOneOf(queryFn.value, [
+            AST_NODE_TYPES.ArrowFunctionExpression,
+            AST_NODE_TYPES.FunctionExpression,
+          ])
         ) {
           return
         }

